components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -702,6 +578,173 @@ public IdentityProvider getResidentIdP(String tenantDomain)
         return identityProvider;
     }
 
+    private String buildSAMLUrl(String urlFromConfigFile, String defaultContext, boolean appendTenantDomainInLegacyMode)
+            throws IdentityProviderManagementServerException {
+
+        String url = urlFromConfigFile;
+        if (StringUtils.isBlank(url)) {
+            url = IdentityUtil.getServerURL(defaultContext, true, true);
```

Comment:
Can't we use the ServiceURLBuilder here to get the URL?

Reply:
Some of the SAML URLs don't need the tenant domain appended as a query param. For example the artifact resolve URL. 

If we use the ServiceURLBuilder all of them will be appended with the tenant domain. 

But i think we can improve and use two methods for the two cases.